src/App.tsx

```diff
@@ -1,24 +1,215 @@
+  const [loadedTodoId, setLoadedTodoId] = useState<number[]>([]);
+  const isError = errorType !== ErrorType.NoError;
+
+  const getTodoList = async () => {
+    try {
+      const newTodoList = await getTodos(USER_ID);
+
+      setTodos(newTodoList);
+    } catch {
+      setErrorType(ErrorType.AddError);
+    }
+  };
+
+  useEffect(() => {
+    if (isError) {
+      setTimeout(() => {
+        setErrorType(ErrorType.NoError);
+      }, 3000);
+    }
+  }, [isError, errorType]);
+
+  useEffect(() => {
+    getTodoList();
+  }, []);
+
+  const filteredTodos = useMemo(() => {
+    switch (status) {
+      case TodoStatus.Active:
+        return todos.filter(todo => !todo.completed);
+
+      case TodoStatus.Completed:
+        return todos.filter(todo => todo.completed);
+
+      default:
+        return todos;
+    }
+  }, [todos, status]);
+
+  const addTodo = async (event: FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (!newTodoTitle.trim()) {
+      setErrorType(ErrorType.EmptyInput);
+
+      return;
+    }
+
+    const newTempTodo = {
+      id: 0,
+      title: newTodoTitle,
+      completed: false,
+    };
+
+    setTempTodo({ ...newTempTodo, userId: USER_ID });
+    setIsLoading(true);
+    setNewTodoTitle('');
+
+    try {
+      const newTodo = await createTodo(USER_ID, newTodoTitle);
+
+      setTodos((prevTodos) => [...prevTodos, newTodo]);
+    } catch {
+      setErrorType(ErrorType.AddError);
+    } finally {
+      setIsLoading(false);
+      setTempTodo(null);
+    }
+  };
+
+  const removeTodo = async (todoId: number) => {
+    try {
+      setLoadedTodoId(state => [...state, todoId]);
+      await deleteTodo(todoId);
+      setTodos((prevTodos) => prevTodos.filter(todo => todo.id !== todoId));
+    } catch {
+      setErrorType(ErrorType.DeleteError);
+    } finally {
+      setLoadedTodoId([]);
+    }
+  };
+
+  const clearCompletedTodos = () => {
+    const completedTodos = todos.filter(todo => todo.completed);
+
+    setLoadedTodoId(completedTodos.map(todo => todo.id));
+
+    completedTodos.forEach(todo => {
+      deleteTodo(todo.id)
+        .then(() => {
+          setTodos(todos.filter(todoItem => !todoItem.completed));
+        })
+        .catch(() => {
+          setErrorType(ErrorType.DeleteError);
+        });
+    });
+  };
+
+  const toggleCompletedTodo = async (todoId: number, completed: boolean) => {
+    setLoadedTodoId(state => [...state, todoId]);
+    try {
+      await updateTodo(todoId, { completed });
+
+      setTodos(state => state.map(todo => {
+        if (todo.id === todoId) {
+          return { ...todo, completed };
+        }
+
+        return todo;
+      }));
+    } catch {
+      setErrorType(ErrorType.UpdateError);
+    } finally {
+      setLoadedTodoId(state => state.filter(todoItem => todoItem !== todoId));
+    }
+  };
+
+  const toggleAllcompletedTodos = () => {
+    const areAllCompleted = todos.every(todo => todo.completed);
+
+    todos.forEach(todo => {
```

Comment:
Instead of using forEach with async operations, consider using Promise.all with map to handle multiple async operations efficiently.